### ISSUE TITLE
google_vertex_ai_featurestore_entitytype_feature: Cannot determine region: set in this resource, or set provider-level 'region' or 'zone'

### ISSUE THREAD

**Issue by shotarok** [hashicorp/terraform-provider-google#13217]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### Affected Resource(s)

* google_vertex_ai_featurestore_entitytype_feature

### Terraform Configuration Files

```tf
resource "google_vertex_ai_featurestore" "featurestore" {
  name   = "tf_update_test"
  region = "us-central1"
}

resource "google_vertex_ai_featurestore_entitytype" "entity" {
  name         = "tf_update_test"
  featurestore = google_vertex_ai_featurestore.featurestore.id
}

resource "google_vertex_ai_featurestore_entitytype_feature" "feature" {
  name       = "tf_update_test"
  entitytype = google_vertex_ai_featurestore_entitytype.entity.id
  value_type = "INT64_ARRAY"
}
```

### Debug Output

<details>

<summary>debug output</summary>

```
2022-12-11T21:10:17.014-0800 [INFO]  Terraform version: 1.1.4
2022-12-11T21:10:17.014-0800 [INFO]  Go runtime version: go1.17.2
2022-12-11T21:10:17.014-0800 [INFO]  CLI args: []string{"/opt/homebrew/Cellar/tfenv/2.2.2/versions/1.1.4/terraform", "apply", "-auto-approve"}
2022-12-11T21:10:17.014-0800 [DEBUG] Attempting to open CLI config file: /Users/shotarok/.terraformrc
2022-12-11T21:10:17.014-0800 [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
2022-12-11T21:10:17.014-0800 [DEBUG] checking for credentials in "/Users/shotarok/.terraform.d/plugins"
2022-12-11T21:10:17.014-0800 [DEBUG] ignoring non-existing provider search directory terraform.d/plugins
2022-12-11T21:10:17.014-0800 [DEBUG] will search for provider plugins in /Users/shotarok/.terraform.d/plugins
2022-12-11T21:10:17.015-0800 [DEBUG] ignoring non-existing provider search directory /Users/shotarok/Library/Application Support/io.terraform/plugins
2022-12-11T21:10:17.015-0800 [DEBUG] ignoring non-existing provider search directory /Library/Application Support/io.terraform/plugins
2022-12-11T21:10:17.016-0800 [INFO]  CLI command args: []string{"apply", "-auto-approve"}
2022-12-11T21:10:17.280-0800 [DEBUG] checking for provisioner in "."
2022-12-11T21:10:17.280-0800 [DEBUG] checking for provisioner in "/opt/homebrew/Cellar/tfenv/2.2.2/versions/1.1.4"
2022-12-11T21:10:17.280-0800 [DEBUG] checking for provisioner in "/Users/shotarok/.terraform.d/plugins"
2022-12-11T21:10:17.282-0800 [INFO]  backend/local: starting Apply operation
2022-12-11T21:10:17.291-0800 [DEBUG] created provider logger: level=debug
2022-12-11T21:10:17.292-0800 [INFO]  provider: configuring client automatic mTLS
2022-12-11T21:10:17.303-0800 [DEBUG] provider: starting plugin: path=.terraform/providers/registry.terraform.io/hashicorp/google/4.45.0/darwin_amd64/terraform-provider-google_v4.45.0_x5 args=[.terraform/providers/registry.terraform.io/hashicorp/google/4.45.0/darwin_amd64/terraform-provider-google_v4.45.0_x5]
2022-12-11T21:10:17.311-0800 [DEBUG] provider: plugin started: path=.terraform/providers/registry.terraform.io/hashicorp/google/4.45.0/darwin_amd64/terraform-provider-google_v4.45.0_x5 pid=67898
2022-12-11T21:10:17.311-0800 [DEBUG] provider: waiting for RPC address: path=.terraform/providers/registry.terraform.io/hashicorp/google/4.45.0/darwin_amd64/terraform-provider-google_v4.45.0_x5
2022-12-11T21:10:17.359-0800 [INFO]  provider.terraform-provider-google_v4.45.0_x5: configuring server automatic mTLS: timestamp=2022-12-11T21:10:17.359-0800
2022-12-11T21:10:17.413-0800 [DEBUG] provider.terraform-provider-google_v4.45.0_x5: plugin address: address=/var/folders/hr/sk3gffkn7p133w3h0vgntx2r0000gn/T/plugin1083270418 network=unix timestamp=2022-12-11T21:10:17.413-0800
2022-12-11T21:10:17.413-0800 [DEBUG] provider: using plugin: version=5
2022-12-11T21:10:17.506-0800 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = transport is closing"
2022-12-11T21:10:17.509-0800 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/hashicorp/google/4.45.0/darwin_amd64/terraform-provider-google_v4.45.0_x5 pid=67898
2022-12-11T21:10:17.509-0800 [DEBUG] provider: plugin exited
2022-12-11T21:10:17.510-0800 [DEBUG] Building and walking validate graph
2022-12-11T21:10:17.514-0800 [DEBUG] ProviderTransformer: "google_vertex_ai_featurestore_entitytype_feature.feature" (*terraform.NodeValidatableResource) needs provider["registry.terraform.io/hashicorp/google"]
2022-12-11T21:10:17.514-0800 [DEBUG] ProviderTransformer: "google_vertex_ai_featurestore.featurestore" (*terraform.NodeValidatableResource) needs provider["registry.terraform.io/hashicorp/google"]
2022-12-11T21:10:17.514-0800 [DEBUG] ProviderTransformer: "google_vertex_ai_featurestore_entitytype.entity" (*terraform.NodeValidatableResource) needs provider["registry.terraform.io/hashicorp/google"]
2022-12-11T21:10:17.514-0800 [DEBUG] ReferenceTransformer: "google_vertex_ai_featurestore.featurestore" references: []
2022-12-11T21:10:17.514-0800 [DEBUG] ReferenceTransformer: "google_vertex_ai_featurestore_entitytype.entity" references: [google_vertex_ai_featurestore.featurestore]
2022-12-11T21:10:17.514-0800 [DEBUG] ReferenceTransformer: "google_vertex_ai_featurestore_entitytype_feature.feature" references: [google_vertex_ai_featurestore_entitytype.entity]
2022-12-11T21:10:17.514-0800 [DEBUG] ReferenceTransformer: "local.gcp_project_id (expand)" references: []
2022-12-11T21:10:17.515-0800 [DEBUG] ReferenceTransformer: "provider[\"registry.terraform.io/hashicorp/google\"]" references: [local.gcp_project_id (expand)]
2022-12-11T21:10:17.515-0800 [DEBUG] Starting graph walk: walkValidate
2022-12-11T21:10:17.518-0800 [DEBUG] created provider logger: level=debug
2022-12-11T21:10:17.518-0800 [INFO]  provider: configuring client automatic mTLS
2022-12-11T21:10:17.526-0800 [DEBUG] provider: starting plugin: path=.terraform/providers/registry.terraform.io/hashicorp/google/4.45.0/darwin_amd64/terraform-provider-google_v4.45.0_x5 args=[.terraform/providers/registry.terraform.io/hashicorp/google/4.45.0/darwin_amd64/terraform-provider-google_v4.45.0_x5]
2022-12-11T21:10:17.535-0800 [DEBUG] provider: plugin started: path=.terraform/providers/registry.terraform.io/hashicorp/google/4.45.0/darwin_amd64/terraform-provider-google_v4.45.0_x5 pid=67899
2022-12-11T21:10:17.535-0800 [DEBUG] provider: waiting for RPC address: path=.terraform/providers/registry.terraform.io/hashicorp/google/4.45.0/darwin_amd64/terraform-provider-google_v4.45.0_x5
2022-12-11T21:10:17.552-0800 [INFO]  provider.terraform-provider-google_v4.45.0_x5: configuring server automatic mTLS: timestamp=2022-12-11T21:10:17.552-0800
2022-12-11T21:10:17.584-0800 [DEBUG] provider: using plugin: version=5
2022-12-11T21:10:17.584-0800 [DEBUG] provider.terraform-provider-google_v4.45.0_x5: plugin address: address=/var/folders/hr/sk3gffkn7p133w3h0vgntx2r0000gn/T/plugin1332955093 network=unix timestamp=2022-12-11T21:10:17.584-0800
2022-12-11T21:10:17.704-0800 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = transport is closing"
2022-12-11T21:10:17.707-0800 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/hashicorp/google/4.45.0/darwin_amd64/terraform-provider-google_v4.45.0_x5 pid=67899
2022-12-11T21:10:17.708-0800 [DEBUG] provider: plugin exited
2022-12-11T21:10:17.708-0800 [INFO]  backend/local: apply calling Plan
2022-12-11T21:10:17.708-0800 [DEBUG] Building and walking plan graph for NormalMode
2022-12-11T21:10:17.709-0800 [DEBUG] ProviderTransformer: "google_vertex_ai_featurestore.featurestore (expand)" (*terraform.nodeExpandPlannableResource) needs provider["registry.terraform.io/hashicorp/google"]
2022-12-11T21:10:17.709-0800 [DEBUG] ProviderTransformer: "google_vertex_ai_featurestore_entitytype.entity (expand)" (*terraform.nodeExpandPlannableResource) needs provider["registry.terraform.io/hashicorp/google"]
2022-12-11T21:10:17.709-0800 [DEBUG] ProviderTransformer: "google_vertex_ai_featurestore_entitytype_feature.feature (expand)" (*terraform.nodeExpandPlannableResource) needs provider["registry.terraform.io/hashicorp/google"]
2022-12-11T21:10:17.709-0800 [DEBUG] ReferenceTransformer: "google_vertex_ai_featurestore.featurestore (expand)" references: []
2022-12-11T21:10:17.709-0800 [DEBUG] ReferenceTransformer: "google_vertex_ai_featurestore_entitytype.entity (expand)" references: [google_vertex_ai_featurestore.featurestore (expand)]
2022-12-11T21:10:17.709-0800 [DEBUG] ReferenceTransformer: "google_vertex_ai_featurestore_entitytype_feature.feature (expand)" references: [google_vertex_ai_featurestore_entitytype.entity (expand)]
2022-12-11T21:10:17.709-0800 [DEBUG] ReferenceTransformer: "local.gcp_project_id (expand)" references: []
2022-12-11T21:10:17.709-0800 [DEBUG] ReferenceTransformer: "provider[\"registry.terraform.io/hashicorp/google\"]" references: [local.gcp_project_id (expand)]
2022-12-11T21:10:17.709-0800 [DEBUG] Starting graph walk: walkPlan
2022-12-11T21:10:17.710-0800 [DEBUG] created provider logger: level=debug
2022-12-11T21:10:17.710-0800 [INFO]  provider: configuring client automatic mTLS
2022-12-11T21:10:17.718-0800 [DEBUG] provider: starting plugin: path=.terraform/providers/registry.terraform.io/hashicorp/google/4.45.0/darwin_amd64/terraform-provider-google_v4.45.0_x5 args=[.terraform/providers/registry.terraform.io/hashicorp/google/4.45.0/darwin_amd64/terraform-provider-google_v4.45.0_x5]
2022-12-11T21:10:17.726-0800 [DEBUG] provider: plugin started: path=.terraform/providers/registry.terraform.io/hashicorp/google/4.45.0/darwin_amd64/terraform-provider-google_v4.45.0_x5 pid=67900
2022-12-11T21:10:17.726-0800 [DEBUG] provider: waiting for RPC address: path=.terraform/providers/registry.terraform.io/hashicorp/google/4.45.0/darwin_amd64/terraform-provider-google_v4.45.0_x5
2022-12-11T21:10:17.744-0800 [INFO]  provider.terraform-provider-google_v4.45.0_x5: configuring server automatic mTLS: timestamp=2022-12-11T21:10:17.743-0800
2022-12-11T21:10:17.775-0800 [DEBUG] provider: using plugin: version=5
2022-12-11T21:10:17.775-0800 [DEBUG] provider.terraform-provider-google_v4.45.0_x5: plugin address: address=/var/folders/hr/sk3gffkn7p133w3h0vgntx2r0000gn/T/plugin2350761542 network=unix timestamp=2022-12-11T21:10:17.775-0800
2022-12-11T21:10:17.892-0800 [WARN]  ValidateProviderConfig from "provider[\"registry.terraform.io/hashicorp/google\"]" changed the config value, but that value is unused
2022-12-11T21:10:17.896-0800 [INFO]  provider.terraform-provider-google_v4.45.0_x5: 2022/12/11 21:10:17 [INFO] Authenticating using DefaultClient...: timestamp=2022-12-11T21:10:17.896-0800
2022-12-11T21:10:17.896-0800 [INFO]  provider.terraform-provider-google_v4.45.0_x5: 2022/12/11 21:10:17 [INFO]   -- Scopes: [https://www.googleapis.com/auth/cloud-platform https://www.googleapis.com/auth/userinfo.email]: timestamp=2022-12-11T21:10:17.896-0800
2022-12-11T21:10:17.896-0800 [INFO]  provider.terraform-provider-google_v4.45.0_x5: 2022/12/11 21:10:17 [INFO] Authenticating using DefaultClient...: timestamp=2022-12-11T21:10:17.896-0800
2022-12-11T21:10:17.896-0800 [INFO]  provider.terraform-provider-google_v4.45.0_x5: 2022/12/11 21:10:17 [INFO]   -- Scopes: [https://www.googleapis.com/auth/cloud-platform https://www.googleapis.com/auth/userinfo.email]: timestamp=2022-12-11T21:10:17.896-0800
2022-12-11T21:10:17.896-0800 [INFO]  provider.terraform-provider-google_v4.45.0_x5: 2022/12/11 21:10:17 [DEBUG] Waiting for state to become: [success]: timestamp=2022-12-11T21:10:17.896-0800
2022-12-11T21:10:18.317-0800 [INFO]  provider.terraform-provider-google_v4.45.0_x5: 2022/12/11 21:10:18 [INFO] Terraform is using this identity: kohama@mercari.com: timestamp=2022-12-11T21:10:18.316-0800
2022-12-11T21:10:18.327-0800 [DEBUG] ReferenceTransformer: "google_vertex_ai_featurestore.featurestore" references: []
google_vertex_ai_featurestore.featurestore: Refreshing state... [id=projects/kouzoh-p-kohama/locations/us-central1/featurestores/tf_update_test]
2022-12-11T21:10:18.334-0800 [INFO]  provider.terraform-provider-google_v4.45.0_x5: 2022/12/11 21:10:18 [DEBUG] Waiting for state to become: [success]: timestamp=2022-12-11T21:10:18.334-0800
2022-12-11T21:10:18.338-0800 [INFO]  provider.terraform-provider-google_v4.45.0_x5: 2022/12/11 21:10:18 [DEBUG] Retry Transport: starting RoundTrip retry loop: timestamp=2022-12-11T21:10:18.338-0800
2022-12-11T21:10:18.338-0800 [INFO]  provider.terraform-provider-google_v4.45.0_x5: 2022/12/11 21:10:18 [DEBUG] Retry Transport: request attempt 0: timestamp=2022-12-11T21:10:18.338-0800
2022-12-11T21:10:18.338-0800 [INFO]  provider.terraform-provider-google_v4.45.0_x5: 2022/12/11 21:10:18 [DEBUG] Google API Request Details:
---[ REQUEST ]---------------------------------------
GET /v1/projects/kouzoh-p-kohama/locations/us-central1/featurestores/tf_update_test?alt=json HTTP/1.1
Host: us-central1-aiplatform.googleapis.com
User-Agent: Terraform/1.1.4 (+https://www.terraform.io) Terraform-Plugin-SDK/2.10.1 terraform-provider-google/dev
Content-Type: application/json
Accept-Encoding: gzip


-----------------------------------------------------: timestamp=2022-12-11T21:10:18.338-0800
2022-12-11T21:10:18.889-0800 [INFO]  provider.terraform-provider-google_v4.45.0_x5: 2022/12/11 21:10:18 [DEBUG] Google API Response Details:
---[ RESPONSE ]--------------------------------------
HTTP/2.0 200 OK
Alt-Svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000,h3-Q050=":443"; ma=2592000,h3-Q046=":443"; ma=2592000,h3-Q043=":443"; ma=2592000,quic=":443"; ma=2592000; v="46,43"
Cache-Control: private
Content-Type: application/json; charset=UTF-8
Date: Mon, 12 Dec 2022 05:10:18 GMT
Server: ESF
Vary: Origin
Vary: X-Origin
Vary: Referer
X-Content-Type-Options: nosniff
X-Frame-Options: SAMEORIGIN
X-Xss-Protection: 0

{
  "name": "projects/415528280434/locations/us-central1/featurestores/tf_update_test",
  "createTime": "2022-12-12T05:03:41.296790Z",
  "updateTime": "2022-12-12T05:03:41.384397Z",
  "etag": "AMEw9yOs2PmUqZrGzk0JWGM3EgqWLADy0UgZ3gbKtSD71FPbv_ZvtEG7DntCUhHiJ2Pl",
  "onlineServingConfig": {},
  "state": "STABLE",
  "streamingIngestionConfig": {
    "mode": "FULL"
  }
}

-----------------------------------------------------: timestamp=2022-12-11T21:10:18.887-0800
2022-12-11T21:10:18.889-0800 [INFO]  provider.terraform-provider-google_v4.45.0_x5: 2022/12/11 21:10:18 [DEBUG] Retry Transport: Stopping retries, last request was successful: timestamp=2022-12-11T21:10:18.887-0800
2022-12-11T21:10:18.889-0800 [INFO]  provider.terraform-provider-google_v4.45.0_x5: 2022/12/11 21:10:18 [DEBUG] Retry Transport: Returning after 1 attempts: timestamp=2022-12-11T21:10:18.887-0800
2022-12-11T21:10:18.902-0800 [WARN]  Provider "registry.terraform.io/hashicorp/google" produced an invalid plan for google_vertex_ai_featurestore.featurestore, but we are tolerating it because it is using the legacy plugin SDK.
    The following problems may be the cause of any confusing errors from downstream operations:
      - .force_destroy: planned value cty.False for a non-computed attribute
      - .labels: planned value cty.MapValEmpty(cty.String) for a non-computed attribute
2022-12-11T21:10:18.905-0800 [DEBUG] ReferenceTransformer: "google_vertex_ai_featurestore_entitytype.entity" references: []
google_vertex_ai_featurestore_entitytype.entity: Refreshing state... [id=projects/kouzoh-p-kohama/locations/us-central1/featurestores/tf_update_test/entityTypes/tf_update_test]
2022-12-11T21:10:18.909-0800 [INFO]  provider.terraform-provider-google_v4.45.0_x5: 2022/12/11 21:10:18 [DEBUG] Waiting for state to become: [success]: timestamp=2022-12-11T21:10:18.909-0800
2022-12-11T21:10:18.909-0800 [INFO]  provider.terraform-provider-google_v4.45.0_x5: 2022/12/11 21:10:18 [DEBUG] Retry Transport: starting RoundTrip retry loop: timestamp=2022-12-11T21:10:18.909-0800
2022-12-11T21:10:18.909-0800 [INFO]  provider.terraform-provider-google_v4.45.0_x5: 2022/12/11 21:10:18 [DEBUG] Retry Transport: request attempt 0: timestamp=2022-12-11T21:10:18.909-0800
2022-12-11T21:10:18.909-0800 [INFO]  provider.terraform-provider-google_v4.45.0_x5: 2022/12/11 21:10:18 [DEBUG] Google API Request Details:
---[ REQUEST ]---------------------------------------
GET /v1/projects/kouzoh-p-kohama/locations/us-central1/featurestores/tf_update_test/entityTypes/tf_update_test?alt=json HTTP/1.1
Host: us-central1-aiplatform.googleapis.com
User-Agent: Terraform/1.1.4 (+https://www.terraform.io) Terraform-Plugin-SDK/2.10.1 terraform-provider-google/dev
Content-Type: application/json
Accept-Encoding: gzip


-----------------------------------------------------: timestamp=2022-12-11T21:10:18.909-0800
2022-12-11T21:10:19.360-0800 [INFO]  provider.terraform-provider-google_v4.45.0_x5: 2022/12/11 21:10:19 [DEBUG] Google API Response Details:
---[ RESPONSE ]--------------------------------------
HTTP/2.0 200 OK
Alt-Svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000,h3-Q050=":443"; ma=2592000,h3-Q046=":443"; ma=2592000,h3-Q043=":443"; ma=2592000,quic=":443"; ma=2592000; v="46,43"
Cache-Control: private
Content-Type: application/json; charset=UTF-8
Date: Mon, 12 Dec 2022 05:10:19 GMT
Server: ESF
Vary: Origin
Vary: X-Origin
Vary: Referer
X-Content-Type-Options: nosniff
X-Frame-Options: SAMEORIGIN
X-Xss-Protection: 0

{
  "name": "projects/415528280434/locations/us-central1/featurestores/tf_update_test/entityTypes/tf_update_test",
  "createTime": "2022-12-12T05:03:49.052029Z",
  "updateTime": "2022-12-12T05:07:11.530212Z",
  "etag": "AMEw9yOTMXfZ_eJfmx6EADNIqD28IlvUoAbD_agYzp_1DvlFriMtyZjkB5r-mzd2XyYe",
  "monitoringConfig": {}
}

-----------------------------------------------------: timestamp=2022-12-11T21:10:19.359-0800
2022-12-11T21:10:19.361-0800 [INFO]  provider.terraform-provider-google_v4.45.0_x5: 2022/12/11 21:10:19 [DEBUG] Retry Transport: Stopping retries, last request was successful: timestamp=2022-12-11T21:10:19.359-0800
2022-12-11T21:10:19.361-0800 [INFO]  provider.terraform-provider-google_v4.45.0_x5: 2022/12/11 21:10:19 [DEBUG] Retry Transport: Returning after 1 attempts: timestamp=2022-12-11T21:10:19.359-0800
2022-12-11T21:10:19.368-0800 [WARN]  Provider "registry.terraform.io/hashicorp/google" produced an invalid plan for google_vertex_ai_featurestore_entitytype.entity, but we are tolerating it because it is using the legacy plugin SDK.
    The following problems may be the cause of any confusing errors from downstream operations:
      - .labels: planned value cty.MapValEmpty(cty.String) for a non-computed attribute
2022-12-11T21:10:19.368-0800 [DEBUG] Resource instance state not found for node "google_vertex_ai_featurestore_entitytype_feature.feature", instance google_vertex_ai_featurestore_entitytype_feature.feature
2022-12-11T21:10:19.369-0800 [DEBUG] ReferenceTransformer: "google_vertex_ai_featurestore_entitytype_feature.feature" references: []
2022-12-11T21:10:19.369-0800 [DEBUG] refresh: google_vertex_ai_featurestore_entitytype_feature.feature: no state, so not refreshing
2022-12-11T21:10:19.376-0800 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = transport is closing"
2022-12-11T21:10:19.376-0800 [DEBUG] provider.terraform-provider-google_v4.45.0_x5: 2022/12/11 21:10:19 [DEBUG] [transport] transport: http2Server.HandleStreams failed to read frame: read unix /var/folders/hr/sk3gffkn7p133w3h0vgntx2r0000gn/T/plugin2350761542->: use of closed network connection
2022-12-11T21:10:19.384-0800 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/hashicorp/google/4.45.0/darwin_amd64/terraform-provider-google_v4.45.0_x5 pid=67900
2022-12-11T21:10:19.384-0800 [DEBUG] provider: plugin exited

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # google_vertex_ai_featurestore_entitytype_feature.feature will be created
  + resource "google_vertex_ai_featurestore_entitytype_feature" "feature" {
      + create_time = (known after apply)
      + entitytype  = "projects/kouzoh-p-kohama/locations/us-central1/featurestores/tf_update_test/entityTypes/tf_update_test"
      + etag        = (known after apply)
      + id          = (known after apply)
      + name        = "tf_update_test"
      + update_time = (known after apply)
      + value_type  = "INT64_ARRAY"
    }

Plan: 1 to add, 0 to change, 0 to destroy.
2022-12-11T21:10:19.387-0800 [INFO]  backend/local: apply calling Apply
2022-12-11T21:10:19.387-0800 [DEBUG] Building and walking apply graph for NormalMode plan
2022-12-11T21:10:19.389-0800 [DEBUG] Resource state not found for node "google_vertex_ai_featurestore_entitytype_feature.feature", instance google_vertex_ai_featurestore_entitytype_feature.feature
2022-12-11T21:10:19.389-0800 [DEBUG] ProviderTransformer: "google_vertex_ai_featurestore.featurestore (expand)" (*terraform.nodeExpandApplyableResource) needs provider["registry.terraform.io/hashicorp/google"]
2022-12-11T21:10:19.389-0800 [DEBUG] ProviderTransformer: "google_vertex_ai_featurestore_entitytype.entity (expand)" (*terraform.nodeExpandApplyableResource) needs provider["registry.terraform.io/hashicorp/google"]
2022-12-11T21:10:19.389-0800 [DEBUG] ProviderTransformer: "google_vertex_ai_featurestore_entitytype_feature.feature (expand)" (*terraform.nodeExpandApplyableResource) needs provider["registry.terraform.io/hashicorp/google"]
2022-12-11T21:10:19.389-0800 [DEBUG] ProviderTransformer: "google_vertex_ai_featurestore_entitytype_feature.feature" (*terraform.NodeApplyableResourceInstance) needs provider["registry.terraform.io/hashicorp/google"]
2022-12-11T21:10:19.389-0800 [DEBUG] ReferenceTransformer: "local.gcp_project_id (expand)" references: []
2022-12-11T21:10:19.390-0800 [DEBUG] ReferenceTransformer: "google_vertex_ai_featurestore_entitytype_feature.feature" references: [google_vertex_ai_featurestore_entitytype.entity (expand)]
2022-12-11T21:10:19.390-0800 [DEBUG] ReferenceTransformer: "provider[\"registry.terraform.io/hashicorp/google\"]" references: [local.gcp_project_id (expand)]
2022-12-11T21:10:19.390-0800 [DEBUG] ReferenceTransformer: "google_vertex_ai_featurestore.featurestore (expand)" references: []
2022-12-11T21:10:19.390-0800 [DEBUG] ReferenceTransformer: "google_vertex_ai_featurestore_entitytype.entity (expand)" references: []
2022-12-11T21:10:19.390-0800 [DEBUG] ReferenceTransformer: "google_vertex_ai_featurestore_entitytype_feature.feature (expand)" references: []
2022-12-11T21:10:19.390-0800 [DEBUG] pruneUnusedNodes: google_vertex_ai_featurestore.featurestore (expand) is no longer needed, removing
2022-12-11T21:10:19.391-0800 [DEBUG] Starting graph walk: walkApply
2022-12-11T21:10:19.391-0800 [DEBUG] created provider logger: level=debug
2022-12-11T21:10:19.391-0800 [INFO]  provider: configuring client automatic mTLS
2022-12-11T21:10:19.403-0800 [DEBUG] provider: starting plugin: path=.terraform/providers/registry.terraform.io/hashicorp/google/4.45.0/darwin_amd64/terraform-provider-google_v4.45.0_x5 args=[.terraform/providers/registry.terraform.io/hashicorp/google/4.45.0/darwin_amd64/terraform-provider-google_v4.45.0_x5]
2022-12-11T21:10:19.413-0800 [DEBUG] provider: plugin started: path=.terraform/providers/registry.terraform.io/hashicorp/google/4.45.0/darwin_amd64/terraform-provider-google_v4.45.0_x5 pid=67902
2022-12-11T21:10:19.413-0800 [DEBUG] provider: waiting for RPC address: path=.terraform/providers/registry.terraform.io/hashicorp/google/4.45.0/darwin_amd64/terraform-provider-google_v4.45.0_x5
2022-12-11T21:10:19.434-0800 [INFO]  provider.terraform-provider-google_v4.45.0_x5: configuring server automatic mTLS: timestamp=2022-12-11T21:10:19.433-0800
2022-12-11T21:10:19.469-0800 [DEBUG] provider: using plugin: version=5
2022-12-11T21:10:19.469-0800 [DEBUG] provider.terraform-provider-google_v4.45.0_x5: plugin address: address=/var/folders/hr/sk3gffkn7p133w3h0vgntx2r0000gn/T/plugin3275896875 network=unix timestamp=2022-12-11T21:10:19.469-0800
2022-12-11T21:10:19.582-0800 [WARN]  ValidateProviderConfig from "provider[\"registry.terraform.io/hashicorp/google\"]" changed the config value, but that value is unused
2022-12-11T21:10:19.584-0800 [INFO]  provider.terraform-provider-google_v4.45.0_x5: 2022/12/11 21:10:19 [INFO] Authenticating using DefaultClient...: timestamp=2022-12-11T21:10:19.584-0800
2022-12-11T21:10:19.584-0800 [INFO]  provider.terraform-provider-google_v4.45.0_x5: 2022/12/11 21:10:19 [INFO]   -- Scopes: [https://www.googleapis.com/auth/cloud-platform https://www.googleapis.com/auth/userinfo.email]: timestamp=2022-12-11T21:10:19.584-0800
2022-12-11T21:10:19.584-0800 [INFO]  provider.terraform-provider-google_v4.45.0_x5: 2022/12/11 21:10:19 [INFO] Authenticating using DefaultClient...: timestamp=2022-12-11T21:10:19.584-0800
2022-12-11T21:10:19.584-0800 [INFO]  provider.terraform-provider-google_v4.45.0_x5: 2022/12/11 21:10:19 [INFO]   -- Scopes: [https://www.googleapis.com/auth/cloud-platform https://www.googleapis.com/auth/userinfo.email]: timestamp=2022-12-11T21:10:19.584-0800
2022-12-11T21:10:19.584-0800 [INFO]  provider.terraform-provider-google_v4.45.0_x5: 2022/12/11 21:10:19 [DEBUG] Waiting for state to become: [success]: timestamp=2022-12-11T21:10:19.584-0800
2022-12-11T21:10:19.999-0800 [INFO]  provider.terraform-provider-google_v4.45.0_x5: 2022/12/11 21:10:19 [INFO] Terraform is using this identity: kohama@mercari.com: timestamp=2022-12-11T21:10:19.999-0800
google_vertex_ai_featurestore_entitytype_feature.feature: Creating...
2022-12-11T21:10:20.013-0800 [INFO]  Starting apply for google_vertex_ai_featurestore_entitytype_feature.feature
2022-12-11T21:10:20.013-0800 [DEBUG] google_vertex_ai_featurestore_entitytype_feature.feature: applying the planned Create change
2022-12-11T21:10:20.016-0800 [ERROR] provider.terraform-provider-google_v4.45.0_x5: Response contains error diagnostic: tf_provider_addr=provider tf_resource_type=google_vertex_ai_featurestore_entitytype_feature tf_rpc=ApplyResourceChange @module=sdk.proto diagnostic_detail= diagnostic_severity=ERROR diagnostic_summary="Cannot determine region: set in this resource, or set provider-level 'region' or 'zone'." tf_proto_version=5.2 tf_req_id=f231761b-9cb8-218d-374e-9874e87a474b @caller=github.com/hashicorp/terraform-plugin-go@v0.10.0/tfprotov5/internal/diag/diagnostics.go:56 timestamp=2022-12-11T21:10:20.016-0800
2022-12-11T21:10:20.027-0800 [ERROR] vertex "google_vertex_ai_featurestore_entitytype_feature.feature" error: Cannot determine region: set in this resource, or set provider-level 'region' or 'zone'.
╷
│ Error: Cannot determine region: set in this resource, or set provider-level 'region' or 'zone'.
│
│   with google_vertex_ai_featurestore_entitytype_feature.feature,
│   on main.tf line 11, in resource "google_vertex_ai_featurestore_entitytype_feature" "feature":
│   11: resource "google_vertex_ai_featurestore_entitytype_feature" "feature" {
│
╵
2022-12-11T21:10:20.039-0800 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = transport is closing"
2022-12-11T21:10:20.040-0800 [DEBUG] provider.terraform-provider-google_v4.45.0_x5: 2022/12/11 21:10:20 [DEBUG] [transport] transport: http2Server.HandleStreams failed to read frame: read unix /var/folders/hr/sk3gffkn7p133w3h0vgntx2r0000gn/T/plugin3275896875->: use of closed network connection
2022-12-11T21:10:20.045-0800 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/hashicorp/google/4.45.0/darwin_amd64/terraform-provider-google_v4.45.0_x5 pid=67902
2022-12-11T21:10:20.045-0800 [DEBUG] provider: plugin exited
```

</details>

### Panic Output

N/A

### Expected Behavior

* FeatureStoreEntityFeature will determine the region from its entity

### Actual Behavior

The following error happens

```
google_vertex_ai_featurestore_entitytype_feature.feature: Creating...
╷
│ Error: Cannot determine region: set in this resource, or set provider-level 'region' or 'zone'.
│
│   with google_vertex_ai_featurestore_entitytype_feature.feature,
│   on main.tf line 11, in resource "google_vertex_ai_featurestore_entitytype_feature" "feature":
│   11: resource "google_vertex_ai_featurestore_entitytype_feature" "feature" {
│
```

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

* There was a similar issue on google_vertex_ai_featurestore_entitytype_feature
  * https://github.com/hashicorp/terraform-provider-google/issues/10100
  * https://github.com/GoogleCloudPlatform/magic-modules/pull/6588/

### References

* b/279524015

**Comment by edwardmedia** [comment]
@shotarok this is dup of https://github.com/hashicorp/terraform-provider-google/issues/12944, a breaking change. Closing as a dup

**Comment by roaks3** [comment]
This issue does look to me like the issue @edwardmedia flagged as a dup. In this case, the `google_vertex_ai_featurestore` sets a `region`, but I believe the current provider implementation requires that the same region is set on the project. In the linked issue, I see the same pattern, but with a `location` field that must match the project's region.

@shotarok can you confirm that setting the region at the project level resolves this issue?

That said, I'm not understanding why this is marked as a breaking change, and I would think that we can improve the ergonomics a bit by using the same pattern as `google_vertex_ai_featurestore_entitytype`, where the region is implied by the parent (that change did not appear to be considered "breaking"). I don't believe users who are already using a project-level region would be affected, assuming that a `google_vertex_ai_featurestore_entitytype_feature` currently cannot exist in a region that is different from its parent.

**Comment by roaks3** [comment]
Reopening because this is still waiting on response from @shotarok , and they have an open PR that might be an improvement. Feel free to close if we are confident this does not need to be addressed, but in that case we should also close the PR (it is over a month old and will start to get stale).

**Comment by rileykarson** [comment]
(watch out when reopening old issues, they're left locked. That doesn't stop maintainers from commenting, but does stop things like other folks commenting and reactions)

**Comment by mahvar** [comment]
Is there an ETA for a fix? 

I'm on google v4.55.0 and still noticing that issue in `google_vertex_ai_endpoint`. Despite the fact that I have set the `location` in the resource, it complains: `Cannot determine region: set in this resource, or set provider-level 'region' or 'zone'.`

I tried to work around it by setting the region in the provider, but I face this: 

```
Error: Provider produced inconsistent result after apply
│ 
│ When applying changes to module.prediction_services["bc-screening"].google_vertex_ai_endpoint.vertex_endpoints["0"], provider
│ "provider[\"registry.terraform.io/hashicorp/google\"]" produced an unexpected new value: Root resource was present, but now absent.
│ 
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
```

**Comment by shotarok** [comment]
> @shotarok can you confirm that setting the region at the project level resolves this issue?

@roaks3 If we set the region in the provider, this issue won't happen. We've been using this as a workaround. Does this answer your question?
